### PR TITLE
[@kadena/react-ui]: Recreating Heading and Text components with VE

### PR DESCRIPTION
**Pull request by eileenmguo**
## Reason:

Recreates the Heading and Text components in the @kadena/react-ui

- Adds VE recipes to implement the styles. Also tested this in a consuming app with RSC (react server components) and the runtime involved with finding the correct classname to apply the styles does not interfere with usage in server components

## Check off the following:

- [x] I have reviewed my changes and run the appropriate tests.
- [ ] I have have run `rush change` to add the appropriate change logs.
- [x] I have added/edited docs.
- [x] I have added tutorials.
- [x] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/libs/react-ui/src/components/Typography/Text/Text.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import {
```
Comment (sstraatemans) [inline packages/libs/react-ui/src/components/Typography/Text/Text.tsx:1]: missing the gradientText component for Typography

File: packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
+/* eslint @typescript-eslint/naming-convention: 0, @kadena-dev/typedef-var: 0 */
```
Comment (sstraatemans) [inline packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts:1]: missing an index file

File: packages/libs/react-ui/src/index.ts [hunk 3]
```diff
@@ -4,9 +4,17 @@ export {
   IIconProps,
   SystemIcon,
   ProductIcon,
+  Text,
```
Comment (sstraatemans) [inline packages/libs/react-ui/src/index.ts:7]: missing the HEader export

**Review by sstraatemans — APPROVED** [review]

**Review by isaozler — APPROVED** [review]

File: packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts (hunk 2 above)
Comment (alber70g) [reply packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts:1]: Why do we need this? I prefer not globally disabling things for a whole file, but using `eslint-disable-next-line` or something similar. If this is a structural problem, we should modify the `@kadena-dev/eslint-config`

**Review by alber70g — APPROVED** [review]

File: packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts (hunk 2 above)
Comment (eileenmguo) [reply packages/libs/react-ui/src/components/Typography/Heading/Heading.css.ts:1]: I just opened another PR to update the @kadena-dev/eslint-config for typedef-var and am going to look into addressing the naming-convention one today 👍

**Pull request merged**